python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```
@@ -1168,117 +1468,277 @@ def _checkSkipped(self, parent, mExposure):
+                # Add a source record for the deconvolved child and
+                # add it to the dictionary of deconvolved parents.
+                deconvolvedParents[parentId] = self._addDeconvolvedParents(
+                    psfParent,
+                    catalog,
+                    deconvolvedFootprints
+                )
+
+        # Add children for all of the deconvolved parents
+        for parentId, sources in deconvolvedParents.items():
+            hierarchy[parentId] = {}
+            for sourceRecord in sources:
+                if (skipArgs := self._checkSkipped(sourceRecord, self.mExposure)) is not None:
+                    self._skipParent(sourceRecord, *skipArgs)
+                    skipped.append(sourceRecord.getId())
+                    continue
+                hierarchy[parentId][sourceRecord.getId()] = _addChildren(sourceRecord)
+
+        return psfParents, hierarchy, skipped
+
+    def _getIntersectingFootprints(
+        self,
+        afwFootprint: afwDet.Footprint,
+        sclFootprints: list[scl.detect.Footprint],
+        footprintImage: scl.Image,
+    ) -> list[afwDet.Footprint]:
+        """Get the intersection of two footprints
+
+        Parameters
+        ----------
+        afwFootprint :
+            The afw footprint
+        sclFootprint :
+            The scarlet footprint
+        footprintImage :
+            An indexed image of the scarlet footprints so that the value
+            of a pixel gives the index + 1 of the footprints that
+            contain that pixel.
+
+        Returns
+        -------
+        intersection :
+            The intersection of the two footprints
+        """
+        footprintIndices = set()
+        ymin, xmin = footprintImage.bbox.origin
+
+        for peak in afwFootprint.peaks:
+            x = peak["i_x"] - xmin
+            y = peak["i_y"] - ymin
+            try:
+                footprintIndex = footprintImage.data[y, x] - 1
+            except:  # noqa: E722
```

What is being caught here? OutOfRange?

---

Yes. If a peak isn't actually contained in the footprint then we just skip it.

---

Ok, but then why bare except instead of a specific exception?

---

This doesn't seem to be resolved.